server/src/main/java/io/crate/analyze/validator/SemanticSortValidator.java

```diff
@@ -91,13 +91,9 @@ public Void visitFunction(Function symbol, SortContext context) {
                                   symbol.valueType())
                 );
             }
-            try {
-                context.inFunction = true;
-                for (Symbol arg : symbol.arguments()) {
-                    arg.accept(this, context);
-                }
-            } finally {
-                context.inFunction = false;
+            context.inFunction = true;
```

Comment:
This fixes `PostgresIT#test_simple_mode_can_handle_npgsql_startup_query` where a case function is used together with `OrderBy`.

Reply:
Hm, not sure about that, the flag is not reset at all now? maybe there is some hidden problem revealed with this.

Reply:
For me at least it doesn't fail with the original version with the `try.... finally`.

Reply:
This modification was based on the previous implementation of the case function. You are right, everything works without this change, with the new version. Thanks for pointing out!